scalyr_agent/copying_manager.py

```diff
@@ -224,6 +226,32 @@ def __init__(self, configuration, monitors):
         StoppableThread.__init__(self, name="log copier thread")
         self.__config = configuration
 
+        self._session = None
+        self._control_plane_client = None
+        self._data_plane_client = None
+        if self.__config.use_new_ingestion:
+            from scalyr_ingestion_client.session import (  # pylint: disable=import-error
+                Session,
+            )
+            from scalyr_ingestion_client.client import (  # pylint: disable=import-error
+                ControlPlaneAPIClient,
+                DataPlaneAPIClient,
+            )
+
+            self._session = Session(uuid=str(uuid.uuid4()))
+
+            self._control_plane_client = ControlPlaneAPIClient(
+                api_token=str(configuration.api_key),
+                cert_path=str(configuration.ca_cert_path),
+            )
+            manager_address = self._control_plane_client.send_client_hello()
```

Comment:
For now doing this here is fine, but in the future we will likely need to handle this elsewhere and also handle retries, etc.

Reply:
Basically, we should do similar as we with the existing client - we use an exponential back off and try to retry connecting to the API.